include/deal.II/lac/generic_linear_algebra.h

```diff
@@ -68,10 +68,51 @@ namespace LinearAlgebraPETSc
 {
   using namespace dealii;
 
+  /**
+   * Typedef for the base class for preconditioners.
+   */
+  typedef PETScWrappers::PreconditionerBase PreconditionerBase;
+
+  /**
+   * Typedef for the base class for linear solvers.
+   */
+  typedef PETScWrappers::SolverBase SolverBase;
```

Comment:
Ah, ok. An example like this ("This typedef is useful in the following context: ...") would be good to have in the documentation.